src/games/strategy/engine/data/Unit.java

```diff
@@ -95,6 +95,13 @@ public boolean equals(final Object o) {
     return this.m_uid.equals(other.m_uid);
   }
 
+  public boolean hasSamePropertiesAndState(final Unit unit) {
```

Comment:
I decided to add this method to Unit itself since its using member fields and can be useful elsewhere.